cmd/update/volume_move.go

```diff
@@ -0,0 +1,383 @@
+	}
+
+	// wait for the new replica on [destination] node to sync
+	err = c.waitForReplicaOnNodeToSync(ctx, nsID)
+	if err != nil {
+		return err
+	}
+
+	c.display.TextMessage(ctx, c.writer, fmt.Sprintf("replica on [destination] node %s finished syncing", c.toNode))
+
+	if !isMovingMaster {
+		_ = c.client.EvictReplica(ctx, nsID.String(), vol.ID.String(), deploymentOnSourceNode.String())
+		if err != nil {
+			if !errors.Is(err, apiclient.VolumeNotFoundError{}) {
+				return fmt.Errorf("failed to force evict replica from [source] node %s (will eventually converge): %w", c.fromNode, err)
+			}
+		}
+
+		c.display.TextMessage(ctx, c.writer, fmt.Sprintf("replica on [source] node %s successfully evicted", c.fromNode))
+		return nil
+	}
+
+	// from here on, we are attempting to move a master
+
+	// find the deployment IDs of the newly created replica and the master being moved
+	var newReplicaID id.Deployment
+	for _, rep := range vol.Replicas {
+		if rep.Node.String() == c.toNode {
+			newReplicaID = rep.ID
+		}
+	}
+	oldMasterID := vol.Master.ID
+
+	err = c.client.AttemptPromotion(ctx, string(nsID), vol.ID.String(), newReplicaID.String())
+	if err != nil {
+		return err
+	}
+
+	c.display.TextMessage(ctx, c.writer, fmt.Sprintf("replica on [destination] node %s attempting promotion to master", c.toNode))
+
+	err = c.waitForMasterPromote(ctx, nsID)
+	if err != nil {
+		return err
+	}
+
+	c.display.TextMessage(ctx, c.writer, fmt.Sprintf("replica on [destination] node %s successfully promoted to master", c.toNode))
+
+	err = c.client.EvictReplica(ctx, nsID.String(), vol.ID.String(), oldMasterID.String())
+	if err != nil {
+		if !errors.Is(err, apiclient.VolumeNotFoundError{}) {
+			return fmt.Errorf("failed to force evict old master from [source] node %s (will eventually converge): %w", oldMasterID.String(), err)
+		}
+	}
+
+	c.display.TextMessage(ctx, c.writer, fmt.Sprintf("old master on [source] node %s successfully evicted", c.toNode))
+	return nil
+}
+
+func (c *volumeMoveCommand) waitForReplicaOnNode(ctx context.Context, nsID id.Namespace) error {
+	for {
+		select {
+		case <-ctx.Done():
+			return fmt.Errorf("timed out while watching for desired state but the operation may still be running (server side)")
+
+		case <-time.After(time.Second * 5):
+			vol, err := c.getVolume(ctx, nsID)
+			if err != nil {
+				return fmt.Errorf("failed to get updated volume config from api: %s", err)
+			}
+
+			// find the new deployment added with this operation
+			for _, rep := range vol.Replicas {
+				if rep.Node.String() == c.toNode {
+					c.display.TextMessage(ctx, c.writer, fmt.Sprintf("new replica added on node %s, starting sync", c.toNode))
+					return nil
+				}
+			}
+		}
+	}
+}
+
+func (c *volumeMoveCommand) waitForReplicaOnNodeToSync(ctx context.Context, nsID id.Namespace) error {
+	for {
+		select {
+		case <-ctx.Done():
+			return fmt.Errorf("timed out while watching for desired state but the operation may still be running (server side)")
```

Comment:
Same point about including `ctx.Err` to this error message

Reply:
Updated as suggested